test/test_quantized.py

```diff
@@ -1055,109 +1055,120 @@ class TestDynamicQuantizedLinear(TestCase):
-            X_q0,
-            X_value_min,
-            X_value_max,
-            W_q0,
-            W_value_min,
-            W_value_max,
-        )
+            if use_multi_dim_input:
+                batch_size *= 3  # Test the multi-dim input tensor
 
-        X_fp32 = torch.from_numpy(_dequantize(X_q0, X_scale, X_zp)).to(dtype=torch.float)
-        if use_multi_dim_input:
-            X_fp32 = X_fp32.view(3, int(batch_size / 3), input_channels)
+            X_scale = 1.0
+            X_zp = 0
+            X_value_min = 0
+            X_value_max = 255
+            X_q0 = np.round(np.random.rand(batch_size, input_channels) *
+                            (X_value_max - X_value_min)
+                            + X_value_min
+                            ).astype(np.uint8)
+            X_q0 = np.round(np.random.rand(batch_size, input_channels) *
+                            (X_value_max - X_value_min) + X_value_min).astype(np.uint8)
+            X_q0[0, 0] = X_value_min
+            X_q0[0, 1] = X_value_max
+
+            # W_scale = 1.0
+            # W_zp = 0
+            W_scales = np.ones(output_channels)
+            W_zps = np.zeros(output_channels).astype(np.int)
+            W_value_min = -128
+            W_value_max = 127
+            W_q0 = np.round(
+                np.random.rand(output_channels, input_channels)
+                * (W_value_max - W_value_min)
+                + W_value_min
+            ).astype(np.int8)
+            W_q0[0, 0] = W_value_min
+            W_q0[1, 0] = W_value_max
 
-        # W_scale, W_zp = _calculate_dynamic_qparams(W_fp32, torch.qint8)
-        # We currently only check the case where W_scale = 1.0, W_zp = 0.
+            b_value_min = -10
+            b_value_max = 10
+            b_q0 = np.round(
+                np.random.rand(output_channels) *
+                (b_value_max - b_value_min) + b_value_min
+            ).astype(np.int32) if use_bias else None
 
-        if use_channelwise:
-            W_fp32 = torch.from_numpy(_dequantize(W_q0, W_scales.reshape(
-                (-1, 1)), W_zps.reshape((-1, 1)))).to(dtype=torch.float)
-            W_q = torch.quantize_per_channel(W_fp32, scales=torch.from_numpy(W_scales),
-                                             zero_points=torch.from_numpy(W_zps), axis=0, dtype=torch.qint8)
-            b_fp32 = torch.from_numpy(
-                _dequantize(b_q0, X_scale * W_scales, 0)
-            ).to(dtype=torch.float) if use_bias else None
-        else:
-            W_fp32 = torch.from_numpy(_dequantize(
-                W_q0, W_scales[0], W_zps[0])).to(dtype=torch.float)
-            W_q = torch.quantize_per_tensor(W_fp32, scale=W_scales[0], zero_point=(
-                W_zps[0].astype(int).item()), dtype=torch.qint8)
-            b_fp32 = torch.from_numpy(
-                _dequantize(b_q0, X_scale * int(W_scales[0].item()), 0)
-            ).to(dtype=torch.float) if use_bias else None
+            avoid_vpmaddubsw_overflow_linear(
```

Comment:
Right: we shouldn't need to run this function (`avoid_vpmaddubsw_overflow_linear`) as there is no saturation issues for QNNPACK raised from `vpmaddubsw` instruction on x86.